cpp/src/lance/io/exec/base.cc

```diff
@@ -32,7 +32,7 @@ ScanBatch ScanBatch::Slice(int64_t off, int64_t length) const {
   auto sliced_batch = batch->Slice(off, length);
   decltype(indices) sliced_indices;
   if (indices) {
-    sliced_indices = std::dynamic_pointer_cast<::arrow::Int32Array>(indices->Slice(offset, length));
+    sliced_indices = std::dynamic_pointer_cast<::arrow::Int32Array>(indices->Slice(off, length));
```

Comment:
This `ScanBatch` struct was a POD struct in the beginning, so that the caller can directly use `scan_batch.batch` to access its fields.  It becomes more complicated lately, we could change it to class and wrap the fields as private and with `_` suffix later?

I am fine to work on it after the release.

Reply:
Added #207 to follow up with the remaining concerns.